app/Models/Bible/BibleBook.php

```diff
@@ -119,4 +137,51 @@ public function book()
     {
         return $this->belongsTo(Book::class);
     }
+
+    /**
+     * Get collection of bible books sorted by bible versification or the book_seq column if it not empty.
+     *
+     * @param string $bible_id
+     * @param string $book_id
+     * @param string $bible_versification
+     *
+     * @return Collection
+     */
+    public static function getAllSortedByversificationOrBookSeq(
+        string $bible_id,
+        ?string $book_id = null,
+        ?string $bible_versification = null
+    ) : Collection {
+        $select_columns = [
+            'bible_books.bible_id',
+            'bible_books.book_id',
+            'bible_books.name',
+            'bible_books.name_short',
+            'bible_books.chapters',
+            'bible_books.book_seq'
+        ];
+
+        if ($bible_versification && Book::isValidVersificationOrderColumn($bible_versification)) {
```

Comment:
calling Book::isValidVersificationOrderColumn is premature... we don't want to execute this in the many cases where the first CASE statement is selected. I would just leave out the check.. we can enforce in the database

Reply:
Ok, I understood. So, I have removed this validation.